### ISSUE TITLE
"Quick Fix" doesn't suggest `__init__.py` imports

### ISSUE THREAD

**Issue by ghost** [microsoft/pylance-release#5231]
With the below setup, when attempting to import `Cat` in a third file, the suggested import path is `from cat.entities import Cat`, rather than `from cat import Cat`. Why are `__init__.py` imports not suggested?

```python
# cat/__init__.py
from cat.entities import Cat

# cat/entities.py
class Cat:
    pass
```

**Comment by heejaechang** [comment]
Currently, we only perform 'alias deduplication' on third-party packages. For user files, we expect packages to be imported directly, without using aliases.

However, we can consider implementing this feature if there is sufficient demand from users. It's important to note that 'alias deduplication' can have performance implications. It makes indexing more resource-intensive. This is especially true as user files are frequently modified, necessitating the re-analysis of all transitive dependencies.

This process is more manageable for third-party packages because they generally do not change after installation, allowing us to analyze and persist the results once.

We cannot apply this optimization to user files since, by their very nature, they are expected to be frequently modified

**Comment by dnsv504** [comment]
@heejaechang thank you for the explanation.

**Comment by debonte** [comment]
@heejaechang, should this be moved to discussions as an enhancement request?

**Comment by debonte** [comment]
This issue was fixed in prerelease version 2024.9.102 and is now also available in stable version 2024.10.1.

Please note that you'll need to set `python.analysis.includeAliasFromUserFiles` to `true` in order to include alias symbols from user files. This will make alias symbols appear in features like Add Import and Auto Import.
